src/main/java/com/minecolonies/coremod/generation/SifterCraftingProvider.java

```diff
@@ -0,0 +1,364 @@
+                        .item(Items.DARK_OAK_SAPLING, 1)
+                        .item(Items.ACACIA_SAPLING, 1)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshDiamond, 5, new LootTableBuilder()
+                        .bonusRolls(0.025f)
+                        .empty(5)
+                        .item(Items.WHEAT_SEEDS, 25)
+                        .item(Items.OAK_SAPLING, 10)
+                        .item(Items.BIRCH_SAPLING, 10)
+                        .item(Items.SPRUCE_SAPLING, 10)
+                        .item(Items.JUNGLE_SAPLING, 10)
+                        .item(Items.CARROT, 5)
+                        .item(Items.POTATO, 5)
+                        .item(Items.PUMPKIN_SEEDS, 5)
+                        .item(Items.MELON_SEEDS, 5)
+                        .item(Items.BEETROOT_SEEDS, 5)
+                        .item(Items.DARK_OAK_SAPLING, 5)
+                        .item(Items.ACACIA_SAPLING, 5)
+                        .build())
+
+                ).collect(Collectors.toList()));
+
+        inputs.put(Items.GRAVEL, Stream.of(
+                new SifterMeshDetails(ModItems.sifterMeshString, 1, new LootTableBuilder()
+                        .empty(85)
+                        .item(Items.IRON_NUGGET, 5)
+                        .item(Items.FLINT, 5)
+                        .item(Items.COAL, 5)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshFlint, 3, new LootTableBuilder()
+                        .empty(60)
+                        .item(Items.IRON_NUGGET, 10)
+                        .item(Items.FLINT, 10)
+                        .item(Items.COAL, 10)
+                        .item(Items.REDSTONE, 10)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshIron, 4, new LootTableBuilder()
+                        .bonusRolls(0.025f)
+                        .empty(46)
+                        .item(Items.REDSTONE, 15)
+                        .item(Items.IRON_NUGGET, 15)
+                        .item(Items.COAL, 15)
+                        .item(Items.LAPIS_LAZULI, 5)
+                        .item(Items.IRON_INGOT, 1)
+                        .item(Items.GOLD_INGOT, 1)
+                        .item(Items.EMERALD, 1)
+                        .item(Items.DIAMOND, 1)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshDiamond, 5, new LootTableBuilder()
+                        .bonusRolls(0.035f)
+                        .empty(40)
+                        .item(Items.REDSTONE, 20)
+                        .item(Items.IRON_NUGGET, 20)
+                        .item(Items.COAL, 20)
+                        .item(Items.LAPIS_LAZULI, 10)
+                        .item(Items.IRON_INGOT, 2)
+                        .item(Items.GOLD_INGOT, 2)
+                        .item(Items.EMERALD, 2)
+                        .item(Items.DIAMOND, 2)

```

Comment:
In a few places I've changed "2.5" from the original json files to "2" because loot table weights are only permitted to be integers.

Reply:
Are you sure about that? I thought I had traced down to the underlying implementation and found it a floating point when I did that.

Reply:
Yes, I'm sure.  The wiki docs for vanilla loot tables specify that both weight and quality are integers, and this also matches the implementation of `ItemLootEntry` in vanilla code and `EntryJson` in the datagen lib.

Reply:
Fair enough